crates/viewer/re_space_view_spatial/src/visualizers/encoded_image.rs

```diff
@@ -42,11 +42,32 @@ impl IdentifiedViewSystem for EncodedImageVisualizer {
     }
 }
 
+struct ImageMediaTypeFilter;
+
+impl VisualizerAdditionalApplicabilityFilter for ImageMediaTypeFilter {
+    /// Marks entities only as applicable for `EncodedImage` if they have an image media type.
+    ///
+    /// Otherwise the image encoder might be suggested for other blobs like video.
+    fn update_applicability(&mut self, event: &re_chunk_store::ChunkStoreEvent) -> bool {
+        diff_component_filter(event, |media_type: &re_types::components::MediaType| {
+            media_type.is_image()
+        }) || diff_component_filter(event, |image: &re_types::components::Blob| {
+            MediaType::guess_from_data(&image.0).map_or(false, |media| media.is_image())
+        })
+    }
```

Comment:
Not new to this PR, but I think there is room for improvement of the naming of these things, maybe:


```suggestion
    fn is_applicable(&mut self, event: &re_chunk_store::ChunkStoreEvent) -> bool {
        is_any_component(event, |media_type: &re_types::components::MediaType| {
            media_type.is_image()
        }) || is_any_component(event, |image: &re_types::components::Blob| {
            MediaType::guess_from_data(&image.0).map_or(false, |media| media.is_image())
        })
    }
```

Also, does this really need `&mut self`?

anyway, no need to fix now. Just thinking out loud.

Reply:
&mut self is for convenience if a trait implementor wants to track additional data that influence appliability